packages/loot-core/src/server/accounts/rules.ts

```diff
@@ -865,21 +865,21 @@ export class RuleIndexer {
     return indexes;
   }
 
-  index(rule) {
+  index(rule: Rule): void {
     const indexes = this.getIndexes(rule);
     indexes.forEach(index => {
       index.add(rule);
     });
   }
 
-  remove(rule) {
+  remove(rule: Rule): void {
     const indexes = this.getIndexes(rule);
     indexes.forEach(index => {
       index.delete(rule);
     });
   }
 
-  getApplicableRules(object) {
+  getApplicableRules(object): Set<Rule> {
```

Comment:
```suggestion
  getApplicableRules(object: Record<string, unknown>): Set<Rule> {
```
(or `Record<string, string | number | string[] | number[]>` as the case may be)

Reply:
I was working on making the whole class generic. So that the type is really enforced. 

The problem I had was that it makes thinks some more verbose while Rule are currently only be used for transactions.  And a lot of the logic inside the file depends on it being a TransactionEnity. 

So I didn't know if it was worth adding.

If you would like to look: https://github.com/UnderKoen/actual/commit/4af6fc8402f83b569fec8dc9a893236cf8e20099

Reply:
Ah, I see. Let's defer to a future PR then!